database/migrations/000065_rule_instance_project_id.up.sql

```diff
@@ -0,0 +1,29 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+BEGIN;
+
+-- add column, leave as not null until we populate the column
+ALTER TABLE rule_instances ADD COLUMN project_id UUID REFERENCES projects(id) ON DELETE CASCADE;
```

Comment:
Do we want or need an index on project_id here?

Reply:
At this point in time, we don't use the new column in the queries since the tuple of (name, profile_id, entity_type) is still unique. As I add more queries, I will add more indices based on common query patterns.